src/Resolvers/Collections.ts

```diff
@@ -92,31 +92,42 @@ export class CollectionsResolver {
 
   @FieldResolver(type => [Collection])
   async relatedCollections(
+    @Arg("size", () => Int, { nullable: true }) size: number,
     @Root() collection: Collection
   ): Promise<Collection[]> {
+    const query: { where: {}; take?: number } = { where: {} }
     const isArtistBased =
       collection.query.artist_ids && collection.query.artist_ids.length > 0
 
+    if (size) {
+      query.take = size + 1
+    }
+
     if (isArtistBased) {
-      const artistResults = await this.repository.find({
-        where: { "query.artist_ids": { $in: collection.query.artist_ids } },
-      })
+      query.where["query.artist_ids"] = {
+        $in: collection.query.artist_ids,
+      }
+
+      const artistResults = await this.repository.find(query)
 
       const relatedArtistCollections = reject(artistResults, {
         id: collection.id,
       })
 
+      // We only want to return relatedArtistCollections when there are more
+      // than 4. Otherwise we return category based related collections.
       if (relatedArtistCollections.length > 4) {
         return relatedArtistCollections
       }
     }
 
-    const relatedCategoryResults = await this.repository.find({
-      where: {
-        category: { $in: [collection.category] },
-        show_on_editorial: true,
-      },
-    })
+    query.where = {
+      category: { $in: [collection.category] },
```

Comment:
Seems like `category` is not array-valued, so I think this could simply be 

```suggestion
      category: collection.category,
```